### PR TITLE
Deploy hypershift + baremetal workers using `none`

### PR DESCRIPTION
**Pull request by e-minguez**
Add instructions to deploy hypershift + add bare metal workers using the `none` provider.

### CODE REVIEW

File: hypershift/hypershift-none.sh [hunk 1]
```diff
@@ -0,0 +1,181 @@
+# Prereqs
+
+# DNS api.hosted0.krnl.es -> each of the nodes (as it is a nodeport)
+# *.apps.hosted0.krnl.es -> IP of the worker
```
Comment (e-minguez) [inline hypershift/hypershift-none.sh]: Or add a load balancer in front...

File: hypershift/hypershift-none.sh [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+EOF
+
+# Type = agent
+envsubst <<"EOF" | oc apply -f -
+apiVersion: hypershift.openshift.io/v1alpha1
+kind: HostedCluster
+metadata:
+  name: ${HOSTED}
+  namespace: ${CLUSTERS_NAMESPACE}
+spec:
```
Comment (e-minguez) [inline hypershift/hypershift-none.sh]: `infrastructureAvailabilityPolicy: SingleReplica`
For a single ingress

Comment (e-minguez) [reply hypershift/hypershift-none.sh]: https://github.com/openshift/hypershift/blob/f8782436393710a1f2acd5dc4b2be5852c799a4d/control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/params.go#L15-L18

File: hypershift/hypershift-none.sh [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    image: "quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE_VERSION}-${OCP_ARCH}"
+EOF
+
+oc extract -n ${CLUSTERS_NAMESPACE} secret/${HOSTED}-admin-kubeconfig --to=- > ${HOSTED}-kubeconfig
+# oc get clusterversion --kubeconfig=${HOSTED}-kubeconfig
+
+# Modify the RHCOS iso to use the ignition generated in the hostedcluster
+mkdir -p iso/
+curl -s -o iso/rhcos-live.x86_64.iso https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.9/4.9.0/rhcos-live.x86_64.iso
+curl -s -k -H "Authorization: Bearer $(oc -n clusters-${HOSTED} get secret $(oc -n clusters-${HOSTED} get secret | grep token-${HOSTED}  | awk '{print $1}') -o jsonpath={.data.token})" https://$(oc get node -o wide | grep master | head -1 | awk '{print $6}'):$(oc -n clusters-${HOSTED} get svc ignition-server -o jsonpath={.spec.ports[0].nodePort})/ignition > iso/worker.ign
```
Comment (e-minguez) [inline hypershift/hypershift-none.sh]: https://github.com/openshift/hypershift/blob/main/ignition-server/main.go#L27

Comment (e-minguez) [reply hypershift/hypershift-none.sh]: Or get the `pointer` ignition instead:

```
oc extract secret/user-data-hosted0-workers-f2a160c1 --keys=value --to=-
```

File: hypershift/baremetal/none.md [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+* DNS entry to point `api.${cluster}.${domain}` to each of the nodes where the hostedcluster will be running. This is because the hosted cluster API is exposed as a `nodeport`. For example:
+
+~~~sh
+api.hosted0.example.com.  IN A  10.19.138.32
+api.hosted0.example.com.  IN A  10.19.138.33
+api.hosted0.example.com.  IN A  10.19.138.37
+~~~
+
+* DNS entry to point `*.apps.${cluster}.${domain}` to a load balancer deployed to redirect incoming traffic to the ingresses pod (the (OpenShift documentation)[https://docs.openshift.com/container-platform/4.9/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-load-balancing-user-infra-example_installing-platform-agnostic] provides some instructions about this)
```
Comment (mvazquezc) [inline hypershift/baremetal/none.md]: (the (OpenShift documentation)[https://docs.openshift.com/container-platform/4.9/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-load-balancing-user-infra-example_installing-platform-agnostic] provides some instructions about this) 

The link is not being rendered properly, probably due to "(the"

File: hypershift/baremetal/none.md [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+  release:
+    image: "quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE_VERSION}-${OCP_ARCH}"
+EOF
+~~~
+
+After a while, a number of pods will be created in the `${CLUSTERS_NAMESPACE}` namespace. Those pods are the control plane of the hosted cluster.
+
+~~~sh
+oc get pods -n ${CLUSTERS_NAMESPACE}
+<insert here>
```
Comment (mvazquezc) [inline hypershift/baremetal/none.md]: Not sure this is intended. Maybe you wanted to add the list of pods?

File: hypershift/baremetal/none.md (hunk 4 above)
Comment (e-minguez) [reply hypershift/baremetal/none.md]: Fixed, thanks!

File: hypershift/baremetal/none.md (hunk 5 above)
Comment (e-minguez) [reply hypershift/baremetal/none.md]: Fixed, thanks!

**Review by mvazquezc — APPROVED** [review]
lgtm

**Pull request merged**